HelloWorld_ST_Sensors/user/gps/gps.cpp

```diff
@@ -0,0 +1,152 @@
+#include "gps.hpp"
+
+namespace gps_ns
+{
+    gps_c::gps_coordinates_s gps_c::get_gps_coordinates(void)
+    {
+        return gps_coordinates_data;
+    }
+
+    void gps_c::gps_struct_init(void)
+    {
+        strcpy(gps_coordinates_data.lat_dir, "NA");
+        strcpy(gps_coordinates_data.latitude, "0.0");
+        strcpy(gps_coordinates_data.lat_dir, "NA");
+        strcpy(gps_coordinates_data.longitude, "0.0");
+        strcpy(gps_coordinates_data.validity, "NA");
+    }
+
+    uint8_t gps_c::check_gprmc_gpgga_line(char *line_buff)
+    {
```

Comment:
Please improve this function, `extract_gpgga_or_gprmc_line` and `gps_interrupt_cb`

Logic should be simple
- Receive one line
- Check if GGA/RMC

- If NO , call registered callback function
- If YES, parse and update lat lon and other data values in the class